src/test/tree/StrategyManangerUnit.tree

```diff
@@ -0,0 +1,89 @@
+StrategyManagerUnit.t.sol
+├── initialize
+│   └── when initialize called again
+│       └── it should revert
+├── depositIntoStrategy()
+│   ├── when deposit successfully
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposit successfully twice
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposits paused
+│   │   └── it should revert
+│   ├── when re-entering
+│   │   └── it should revert
+│   ├── when safeTransferFrom() function reverts
+│   │   └── it should revert
+│   ├── when IERC20 token does not exist
+│   │   └── it should revert
+│   ├── when strategy.deposit() function reverts
+│   │   └── it should revert
+│   ├── when strategy does not exist
+│   │   └── it should revert
+│   ├── when strategy is not whitelisted
+│   │   └── it should revert
+│   └── _addShares()
+│       ├── when shares is 0
+│       │   └── it should revert
+│       └── when deposit would exceed max array length
+│           └── it should revert
+├── depositIntoStrategyWithSignature()
+│   ├── when deposit with signature successfully
+│   │   └── staker nonce should increase and shares should increase
+│   ├── when deposit with signature successfully and replay same deposit signature
+│   │   └── it should revert the second time
+│   ├── when deposit with signature with EIP1271 compliant contract signature
+│   │   └── it should deposit successfully with shares and nonce increase
+│   ├── when deposit with signature with EIP1271 compliant contract but with bad signature
+│   │   └── it should revert
+│   ├── when deposit with signature with non-compliant EIP1271 contract
+│   │   └── it should revert
+│   ├── when deposits paused
+│   │   └── it should revert
+│   ├── when re-entering
+│   │   └── it should revert
+│   ├── when signature expired
```

Comment:
Personally prefer for #1 to simply have the external function name for better readability and makes more logical sense since we are trying to cover all the possible code paths from an external user. Will change #2 and #3 to be more consistent

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/293/commits/3b6a00fa9b6b51a6cf1705b5e0e2f6e0c4fd6581